### PR TITLE
Make the usage of Ftactic in Tacinterp more consistent

### PR DESCRIPTION
**Pull request by afdw**
This file uses a lot shadowing, which makes it hard to understand.

Suffix top-level functions that return `_ Ftactic.t` with "_ftactic" instead of using shadowing (this way the external interface is preserved).

Also clean up control flow and local shadowing in `eval_tactic_ist`, `interp_tacarg_ftactic` and `interp_message_ftactic`.

Make the usage of `Ftactic.bind` consistent, factorize the usage of `Ftactic.run` as much as possible.

### CODE REVIEW

**Comment by afdw** [comment]
> I'm not super fan of the `_ftactic` nomenclature, this looks heavyweight. I'd rather favour a one-letter prefix `f` or something similar.

`_f` is what I considered going with first, but I thought that it can be hard to understand what "f" is supposed to mean.

**Comment by silene** [comment]
It seems like most of the changes are just spurious reformatting. This makes it hard to review.

**Comment by afdw** [comment]
@ppedrot @silene I have split into commits and removed purely formatting changes.

File: plugins/ltac/tacinterp.ml [hunk 1]
```diff
@@ -1192,51 +1192,44 @@ and eval_tactic_ist ist tac : unit Proofview.tactic =
...
   | TacML (opn,l) ->
       let trace = push_trace (Loc.tag ?loc @@ LtacMLCall tac) ist in
-      let ist = { ist with extra = TacStore.set ist.extra f_trace trace; } in
+      let ist = {ist with extra = TacStore.set ist.extra f_trace trace} in
+      let lr_f = l |> Ftactic.List.map_right (interp_tacarg ist) in
       let tac = Tacenv.interp_ml_tactic opn in
-      let args = Ftactic.List.map_right (fun a -> interp_tacarg ist a) l in
-      let tac args =
-        let name () = Pptactic.pr_extend print_top_val 0 opn args in
+      Ftactic.run lr_f (fun lr ->
```
Comment (ppedrot) [inline plugins/ltac/tacinterp.ml:1227]: I personally don't like this style, I explicitly prefer having intermediate named values and closures.

Comment (afdw) [reply plugins/ltac/tacinterp.ml]: I have switched the style to name continuations as variables.

Comment (SkySkimmer) [reply plugins/ltac/tacinterp.ml]: TBH I preferred it with the anonymous lambda

Comment (SkySkimmer) [reply plugins/ltac/tacinterp.ml]: but feel free to ignore me on this

Comment (afdw) [reply plugins/ltac/tacinterp.ml]: My intention was to make use of fewer intermediate variables (which are hard to keep track of, and require back and forth when reading code) and to make the code as "linear" and simple as possible.

Comment (ppedrot) [reply plugins/ltac/tacinterp.ml]: I guess that the last one to modify the code keeps the right to leave it as they want. That said, I'm probably one of the main maintainers of Ltac1 so I'd also argue for the old style. Anyways this is not really a strong preference, it's just a matter of (arguable) taste.

**Comment by herbelin** [comment]
I'm afraid that I have no idea about how to implement them without having first to spend some time at how the module works. Is there a canonical way to encapsulate the underlying Set functions without thinking too hard?

**Comment by herbelin** [comment]
Sorry, last comment is on wrong PR. Deleting it.

**Pull request closed without merging**